--- v8/src/cpu.cc       2014-02-11 13:02:05.810134031 +0400
+++ v8/src/cpu.cc       2014-02-11 13:04:09.118128534 +0400
@@ -379,7 +379,12 @@ CPU::CPU() : stepping_(0),
     // an ARMv6-one. For example, the Raspberry Pi is one popular
     // ARMv6 device that reports architecture 7.
     if (architecture_ == 7) {
-      char* processor = cpu_info.ExtractField("Processor");
+      char* processor;
+      processor = cpu_info.ExtractField("Processor");
+      if (HasListItem(processor, "(v6l)")) {
+        architecture_ = 6;
+      }
+      processor = cpu_info.ExtractField("model name");
       if (HasListItem(processor, "(v6l)")) {
         architecture_ = 6;
       }
 
