--- v8/build/standalone.gypi.orig	2015-04-21 19:37:49.000000000 +0200
+++ v8/build/standalone.gypi	2015-04-21 19:17:36.000000000 +0200
@@ -105,6 +105,7 @@
         (v8_target_arch=="arm64" and host_arch!="arm64") or \
         (v8_target_arch=="mipsel" and host_arch!="mipsel") or \
         (v8_target_arch=="x64" and host_arch!="x64") or \
+        (v8_target_arch!="x64" and host_arch=="x64") or \
         (OS=="android" or OS=="qnx")', {
         'want_separate_host_toolset': 1,
       }, {
